diff --git a/app/decorators/models/solidus_easypost/spree/address_decorator.rb b/app/decorators/models/solidus_easypost/spree/address_decorator.rb
new file mode 100644
index 0000000..d122bd3
--- /dev/null
+++ b/app/decorators/models/solidus_easypost/spree/address_decorator.rb
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module SolidusEasypost
+  module Spree
+    module EasyPost
+      module AddressDecorator
+        def easypost_address
+          attributes = {
+            street1: address1,
+            street2: address2,
+            city: city,
+            zip: zipcode,
+            phone: phone
+          }
+
+          attributes[:company] = company if respond_to?(:company)
+          attributes[:name] = full_name if respond_to?(:full_name)
+          attributes[:state] = state ? state.abbr : state_name
+          attributes[:country] = country.try(:iso)
+
+          ::EasyPost::Address.create attributes
+        end
+
+        ::Spree::Address.prepend self
+        ::Spree::StockLocation.prepend self
+      end
+    end
+  end
+end
diff --git a/app/decorators/models/solidus_easypost/spree/shipment_decorator.rb b/app/decorators/models/solidus_easypost/spree/shipment_decorator.rb
new file mode 100644
index 0000000..4d10c53
--- /dev/null
+++ b/app/decorators/models/solidus_easypost/spree/shipment_decorator.rb
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module SolidusEasypost
+  module Spree
+    module ShipmentDecorator
+      def self.prepended(mod)
+        mod.state_machine.before_transition(
+          to: :shipped,
+          do: :buy_easypost_rate,
+          if: -> { Solidus::EasyPost::CONFIGS[:purchase_labels?] }
+        )
+      end
+
+      def easypost_shipment
+        if selected_easy_post_shipment_id
+          @ep_shipment ||= ::EasyPost::Shipment.retrieve(selected_easy_post_shipment_id)
+        else
+          @ep_shipment = build_easypost_shipment
+        end
+      end
+
+      private
+
+      def selected_easy_post_rate_id
+        selected_shipping_rate.easy_post_rate_id
+      end
+
+      def selected_easy_post_shipment_id
+        selected_shipping_rate.easy_post_shipment_id
+      end
+
+      def build_easypost_shipment
+        ::EasyPost::Shipment.create(
+          to_address: order.ship_address.easypost_address,
+          from_address: stock_location.easypost_address,
+          parcel: to_package.easypost_parcel
+        )
+      end
+
+      def buy_easypost_rate
+        rate = easypost_shipment.rates.find do |sr|
+          sr.id == selected_easy_post_rate_id
+        end
+
+        easypost_shipment.buy(rate)
+        self.tracking = easypost_shipment.tracking_code
+      end
+
+      ::Spree::Shipment.prepend self
+    end
+  end
+end
diff --git a/app/decorators/models/solidus_easypost/spree/shipping_rate_decorator.rb b/app/decorators/models/solidus_easypost/spree/shipping_rate_decorator.rb
new file mode 100644
index 0000000..404ba46
--- /dev/null
+++ b/app/decorators/models/solidus_easypost/spree/shipping_rate_decorator.rb
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module SolidusEasypost
+  module Spree
+    module ShippingRateDecorator
+      def name
+        read_attribute(:name) || super
+      end
+
+      ::Spree::ShippingRate.prepend self
+    end
+  end
+end
diff --git a/app/decorators/models/solidus_easypost/spree/stock/estimator_decorator.rb b/app/decorators/models/solidus_easypost/spree/stock/estimator_decorator.rb
new file mode 100644
index 0000000..c4067b7
--- /dev/null
+++ b/app/decorators/models/solidus_easypost/spree/stock/estimator_decorator.rb
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module SolidusEasypost
+  module Spree
+    module Stock
+      module EstimatorDecorator
+        def shipping_rates(package, frontend_only = true)
+          SolidusEasypost::Estimator.new.shipping_rates(package, frontend_only)
+        end
+
+        ::Spree::Stock::Estimator.prepend self
+      end
+    end
+  end
+end
diff --git a/app/decorators/models/solidus_easypost/spree/stock/package_decorator.rb b/app/decorators/models/solidus_easypost/spree/stock/package_decorator.rb
new file mode 100644
index 0000000..8b3d939
--- /dev/null
+++ b/app/decorators/models/solidus_easypost/spree/stock/package_decorator.rb
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module SolidusEasypost
+  module Spree
+    module Stock
+      module PackageDecorator
+        def easypost_parcel
+          total_weight = contents.sum do |item|
+            item.quantity * item.variant.weight
+          end
+
+          ::EasyPost::Parcel.create weight: total_weight
+        end
+
+        def easypost_shipment
+          ::EasyPost::Shipment.create(
+            to_address: order.ship_address.easypost_address,
+            from_address: stock_location.easypost_address,
+            parcel: easypost_parcel
+          )
+        end
+
+        ::Spree::Stock::Package.prepend self
+      end
+    end
+  end
+end
diff --git a/app/models/solidus_easypost/estimator.rb b/app/models/solidus_easypost/estimator.rb
index 853f75b..48f9d34 100644
--- a/app/models/solidus_easypost/estimator.rb
+++ b/app/models/solidus_easypost/estimator.rb
@@ -10,7 +10,7 @@ module SolidusEasypost
 
       if rates.any?
         rates.each do |rate|
-          spree_rate = Spree::ShippingRate.new(
+          spree_rate = ::Spree::ShippingRate.new(
             name: "#{rate.carrier} #{rate.service}",
             cost: rate.rate,
             easy_post_shipment_id: rate.shipment_id,
@@ -39,12 +39,12 @@ module SolidusEasypost
     # and should not be changed in the admin.
     def find_or_create_shipping_method(rate)
       method_name = "#{rate.carrier} #{rate.service}"
-      Spree::ShippingMethod.find_or_create_by(admin_name: method_name) do |r|
+      ::Spree::ShippingMethod.find_or_create_by(admin_name: method_name) do |r|
         r.name = method_name
         r.available_to_users = false
         r.code = rate.service
-        r.calculator = Spree::Calculator::Shipping::FlatRate.create
-        r.shipping_categories = [Spree::ShippingCategory.first]
+        r.calculator = ::Spree::Calculator::Shipping::FlatRate.create
+        r.shipping_categories = [::Spree::ShippingCategory.first]
       end
     end
   end
diff --git a/app/models/spree/address_decorator.rb b/app/models/spree/address_decorator.rb
deleted file mode 100644
index 1b28271..0000000
--- a/app/models/spree/address_decorator.rb
+++ /dev/null
@@ -1,27 +0,0 @@
-# frozen_string_literal: true
-
-module Spree
-  module EasyPost
-    module AddressDecorator
-      def easypost_address
-        attributes = {
-          street1: address1,
-          street2: address2,
-          city: city,
-          zip: zipcode,
-          phone: phone
-        }
-
-        attributes[:company] = company if respond_to?(:company)
-        attributes[:name] = full_name if respond_to?(:full_name)
-        attributes[:state] = state ? state.abbr : state_name
-        attributes[:country] = country.try(:iso)
-
-        ::EasyPost::Address.create attributes
-      end
-    end
-  end
-end
-
-Spree::Address.prepend Spree::EasyPost::AddressDecorator
-Spree::StockLocation.prepend Spree::EasyPost::AddressDecorator
diff --git a/app/models/spree/package_decorator.rb b/app/models/spree/package_decorator.rb
deleted file mode 100644
index 2f35b67..0000000
--- a/app/models/spree/package_decorator.rb
+++ /dev/null
@@ -1,25 +0,0 @@
-# frozen_string_literal: true
-
-module Spree
-  module Stock
-    module PackageDecorator
-      def easypost_parcel
-        total_weight = contents.sum do |item|
-          item.quantity * item.variant.weight
-        end
-
-        ::EasyPost::Parcel.create weight: total_weight
-      end
-
-      def easypost_shipment
-        ::EasyPost::Shipment.create(
-          to_address: order.ship_address.easypost_address,
-          from_address: stock_location.easypost_address,
-          parcel: easypost_parcel
-        )
-      end
-    end
-  end
-end
-
-Spree::Stock::Package.prepend Spree::Stock::PackageDecorator
diff --git a/app/models/spree/shipment_decorator.rb b/app/models/spree/shipment_decorator.rb
deleted file mode 100644
index d1d462c..0000000
--- a/app/models/spree/shipment_decorator.rb
+++ /dev/null
@@ -1,50 +0,0 @@
-# frozen_string_literal: true
-
-module Spree
-  module ShipmentDecorator
-    def self.prepended(mod)
-      mod.state_machine.before_transition(
-        to: :shipped,
-        do: :buy_easypost_rate,
-        if: -> { Solidus::EasyPost::CONFIGS[:purchase_labels?] }
-      )
-    end
-
-    def easypost_shipment
-      if selected_easy_post_shipment_id
-        @ep_shipment ||= ::EasyPost::Shipment.retrieve(selected_easy_post_shipment_id)
-      else
-        @ep_shipment = build_easypost_shipment
-      end
-    end
-
-    private
-
-    def selected_easy_post_rate_id
-      selected_shipping_rate.easy_post_rate_id
-    end
-
-    def selected_easy_post_shipment_id
-      selected_shipping_rate.easy_post_shipment_id
-    end
-
-    def build_easypost_shipment
-      ::EasyPost::Shipment.create(
-        to_address: order.ship_address.easypost_address,
-        from_address: stock_location.easypost_address,
-        parcel: to_package.easypost_parcel
-      )
-    end
-
-    def buy_easypost_rate
-      rate = easypost_shipment.rates.find do |sr|
-        sr.id == selected_easy_post_rate_id
-      end
-
-      easypost_shipment.buy(rate)
-      self.tracking = easypost_shipment.tracking_code
-    end
-  end
-end
-
-Spree::Shipment.prepend Spree::ShipmentDecorator
diff --git a/app/models/spree/shipping_rate_decorator.rb b/app/models/spree/shipping_rate_decorator.rb
deleted file mode 100644
index d59a1e2..0000000
--- a/app/models/spree/shipping_rate_decorator.rb
+++ /dev/null
@@ -1,11 +0,0 @@
-# frozen_string_literal: true
-
-module Spree
-  module ShippingRateDecorator
-    def name
-      read_attribute(:name) || super
-    end
-  end
-end
-
-Spree::ShippingRate.prepend Spree::ShippingRateDecorator
diff --git a/app/models/spree/stock/estimator_decorator.rb b/app/models/spree/stock/estimator_decorator.rb
deleted file mode 100644
index b25f5ed..0000000
--- a/app/models/spree/stock/estimator_decorator.rb
+++ /dev/null
@@ -1,13 +0,0 @@
-# frozen_string_literal: true
-
-module Spree
-  module Stock
-    module EstimatorDecorator
-      def shipping_rates(package, frontend_only = true)
-        SolidusEasypost::Estimator.new.shipping_rates(package, frontend_only)
-      end
-    end
-  end
-end
-
-Spree::Stock::Estimator.prepend Spree::Stock::EstimatorDecorator
